lib/ansible/galaxy/dependency_resolution/__init__.py

```diff
@@ -45,5 +49,6 @@ def build_collection_dependency_resolver(
             upgrade=upgrade,
             include_signatures=include_signatures,
         ),
-        CollectionDependencyReporter(),
+        CollectionDependencyDebuggingReporter() if C.DEFAULT_DEBUG
```

Comment:
Could we add a dedicated config for the ansible-galaxy reporter mode, so its easy to add new reporters in the future?

Reply:
Should we still consult the debug mode along with that?